microcosm_flask/profiling.py

```diff
@@ -1,7 +1,14 @@
+from datetime import datetime
 from os import makedirs
-from os.path import exists, expanduser
+from os.path import exists, expanduser, join
 
-from werkzeug.contrib.profiler import ProfilerMiddleware
+from flask import g
+
+
+try:
+    from pyinstrument import Profiler
+except ImportError:
+    raise Exception("Profiling requires 'pyinstrument'.")
```

Comment:
the purpose with catching import error is actually to swallow the exception so that if you dont have the extras_required, just warn the user if at all. so in this case maybe as simple as:

```python
...
except ImportError:
    pass
```

and then can add a check below inside the function along lines of:

```python
...
  if not getattr(g, 'profiler', None):
    raise Exception("profiler is not installed. make sure pyinstrument is installed")
```

Reply:
I think it is good to go now @adamhadani !